test/RewardsStreamerMP.t.sol

```diff
@@ -1989,3 +1924,121 @@ contract MaliciousUpgradeTest is RewardsStreamerMPTest {
         assertEq(stakingToken.balanceOf(alice), aliceInitialBalance, "Alice should get her tokens back");
     }
 }
+
+contract RewardsStreamerMP_RewardsTest is RewardsStreamerMPTest {
+    function setUp() public virtual override {
+        super.setUp();
+    }
+
+    function testSetRewards() public {
+        assertEq(streamer.rewardStartTime(), 0);
+        assertEq(streamer.rewardEndTime(), 0);
+        assertEq(streamer.lastRewardTime(), 0);
+
+        uint256 currentTime = vm.getBlockTimestamp();
+        // just to be sure that currentTime is not 0
+        // since we are testing that it is used for rewardStartTime
+        currentTime += 1 days;
+        vm.warp(currentTime);
+        vm.prank(admin);
+        streamer.setReward(1000, 10);
+
+        assertEq(streamer.rewardStartTime(), currentTime);
+        assertEq(streamer.rewardEndTime(), currentTime + 10);
+        assertEq(streamer.lastRewardTime(), currentTime);
+    }
+
+    function testSetRewards_RevertsNotAuthorized() public {
+        vm.prank(alice);
+        vm.expectPartialRevert(Ownable.OwnableUnauthorizedAccount.selector);
+        streamer.setReward(1000, 10);
+    }
+
+    function testSetRewards_RevertsBadDuration() public {
+        vm.prank(admin);
+        vm.expectRevert(RewardsStreamerMP.StakingManager__DurationCannotBeZero.selector);
+        streamer.setReward(1000, 0);
+    }
+
+    function testSetRewards_RevertsBadAmount() public {
+        vm.prank(admin);
+        vm.expectRevert(RewardsStreamerMP.StakingManager__AmountCannotBeZero.selector);
+        streamer.setReward(0, 10);
+    }
+
+    function testTotalRewardsSupply() public {
+        _stake(alice, 100e18, 0);
+        assertEq(streamer.totalRewardsSupply(), 0);
+
+        uint256 initialTime = vm.getBlockTimestamp();
+
+        vm.prank(admin);
+        streamer.setReward(1000e18, 10 days);
+        assertEq(streamer.totalRewardsSupply(), 0);
+
+        for (uint256 i = 0; i <= 10; i++) {
+            vm.warp(initialTime + i * 1 days);
+            assertEq(streamer.totalRewardsSupply(), 100e18 * i);
+        }
+
+        // after the end of the reward period, the total rewards supply does not increase
+        vm.warp(initialTime + 11 days);
+        assertEq(streamer.totalRewardsSupply(), 1000e18);
+        assertEq(streamer.totalRewardsAccrued(), 0);
+
+        uint256 secondRewardTime = initialTime + 20 days;
+        vm.warp(secondRewardTime);
+
+        // still the same rewards supply after 20 days
+        assertEq(streamer.totalRewardsSupply(), 1000e18);
+        assertEq(streamer.totalRewardsAccrued(), 0);
+
+        // set other 2000 rewards for other 10 days
+        vm.prank(admin);
+        streamer.setReward(2000e18, 10 days);
+        // accrued is 1000 from the previous reward and still 0 for the new one
+        assertEq(streamer.totalRewardsSupply(), 1000e18, "totalRewardsSupply should be 1000");
+        assertEq(streamer.totalRewardsAccrued(), 1000e18);
```

Comment:
Ah, interesting. Looking at these assertions, I expected it to be the other way around:

- `totalRewardsSupply()` is what is "checkpointed"
- `totalRewardsAccrued()` is "real-time" balance that is accrued until this point in time

Also, the comment says `[...] and still 0 for the new one` but the `0` isn't reflected here anywhere.
Likely needs rewording?

Reply:
@0x-r4bbit yes so we have:

* 1000 accrued and saved to storage in `totalRewardsAccrued`
* 0 pending
* `totalRewardsSupply` is `totalRewardsAccrued` + pending = `1000`